objectivec/include/ort_session.h

```diff
@@ -46,14 +54,151 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @param inputs Dictionary of input names to input ORT values.
  * @param outputNames Set of output names.
+ * @param runOptions Run configuration options.
  * @param[out] error Optional error information set if an error occurs.
  * @return A dictionary of output names to output ORT values with the outputs
  *         requested in `outputNames`, or nil if an error occurs.
  */
 - (nullable NSDictionary<NSString*, ORTValue*>*)runWithInputs:(NSDictionary<NSString*, ORTValue*>*)inputs
                                                   outputNames:(NSSet<NSString*>*)outputNames
+                                                   runOptions:(ORTRunOptions*)runOptions
                                                         error:(NSError**)error;
 
 @end
 
+/**
+ * Options for configuring a session.
+ */
+@interface ORTSessionOptions : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates session configuration options.
+ *
+ * @param[out] error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithError:(NSError**)error NS_SWIFT_NAME(init());
+
+/**
+ * Sets the number of threads used to parallelize the execution within nodes.
+ * A value of 0 means ORT will pick a default value.
+ *
+ * @param intraOpNumThreads The number of threads.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setIntraOpNumThreads:(int)intraOpNumThreads
+                       error:(NSError**)error;
+
+/**
+ * Sets the graph optimization level.
+ *
+ * @param graphOptimizationLevel The graph optimization level.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setGraphOptimizationLevel:(ORTGraphOptimizationLevel)graphOptimizationLevel
+                            error:(NSError**)error;
+
+/**
+ * Sets the path to which the optimized model file will be saved.
+ *
+ * @param optimizedModelFilePath The optimized model file path.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setOptimizedModelFilePath:(NSString*)optimizedModelFilePath
+                            error:(NSError**)error;
+
```

Comment:
Is there any scenario where someone would be saving an optimized model on a phone? Or would this API be potentially used on say a Mac laptop?

Reply:
I suppose this might be more useful on MacOS.